python/tank/bootstrap/manager.py

```diff
@@ -221,11 +244,50 @@ def set_progress_callback(self, callback):
         """
         self._progress_cb = callback
 
-    def bootstrap_engine(self, engine_name, entity=None):
+    def bootstrap_engine(self,
```

Comment:
Regarding the progress callback, when bootstrapping an Sgtk instance asynchronously, method (not property) `ToolkitManager.set_progress_callback()` cannot be used to set the progress callback from the client code.

In this case, ToolkitManager progress callback is set to `BootstrapWorker._report_progress()` and the following callback chain is used:

`ToolkitManager._report_progress()`
--> `BootstrapWorker._report_progress()`
--> signal `BootstrapWorker.progressing`
--> slot `BootstrapSupervisor._progress_bootstrap()`
--> client progress callback